app/vector_icons/BUILD.gn

```diff
@@ -7,30 +7,53 @@ import("//brave/components/speedreader/common/buildflags/buildflags.gni")
 import("//build/config/ui.gni")
 import("//components/vector_icons/vector_icons.gni")
 
+action_foreach("leo_icons") {
+  script = "//brave/app/vector_icons/convert_svg_to_icon.py"
+
+  # These should be the names Leo Icons
+  # They will be available in C++ as |kLeo{IconName}Icon|
+  icons = [
+    "arrow-right.svg",
+    "heart-filled.svg",
+    "heart-outline.svg",
+    "rss.svg",
+  ]
+
+  leo_icons_base_path = "../../node_modules/@brave/leo/icons"
+  sources = rebase_path(icons, ".", leo_icons_base_path)
+
+  outputs = [ "$target_gen_dir/icons/leo_{{source_name_part}}.icon" ]
+  args = [
+    "{{source}}",
+    rebase_path(target_gen_dir, root_build_dir) +
+        "/icons/leo_{{source_name_part}}.icon",
+  ]
+}
+
 aggregate_vector_icons("brave_vector_icons") {
   icon_directory = "."
 
+  leo_icons = get_target_outputs(":leo_icons")
```

Comment:
I'm wondering what if we could make a Leo component and have a component/leo/vector_icons.h instead. That could be more bothersome, but we might be able to remove alt_icons_dir patching. Also, if we're going to make Leo more general components , I believe leo should be go to lower layer like views or components. So having separate BUILD target / and aggregated vector_icons.h would be better for that. I'd like to hear your take on this 😄

Reply:
Yeah, originally I was just going to compile all of the SVGs to `.icon` files in the Leo repository (potentially as a build step). However I thought it might make more sense to do this in Brave Core, as it's probably the only place which will use the Skia icons :laughing: Happy to move it to Leo though, which would solve the patching issue.

I don't think we want to generate an `aggregated_vector_icons.h` file in Leo, just so we can pick only the icons we need (rather than compiling all ~1700 of them). Sorry if that's not what you meant :D

Reply:
My comment was not correct :) I believe we're on the same page. I imagined something like "components/leo/vector_icons.h" which only contains leo icons.